TheMovieDb/Network/APIService.swift

```diff
@@ -0,0 +1,50 @@
+//
+//  APIService.swift
+//  TheMovieDb
+//
+//  Created by Rob Cruz on 07/11/21.
+//
+
+import Foundation
+
+class APIService {
+
+    private var dataTask: URLSessionDataTask?
+    
+    func getMoviesData(url: String, completion: @escaping (Result<MoviesData, Error>) -> Void) {
+        
+        guard URL(string: url) != nil else { return }
+        
+        dataTask = URLSession.shared.dataTask(with: url) { (data, response, error) in
```

Comment:
Holding the reference of this task is not need it. (unless you meant to stop the previous task if not nil)

[Functional programming - Pure functions](https://www.swiftbysundell.com/articles/pure-functions-in-swift/)